lib/sock_js/sock_js_utils.dart

```diff
@@ -0,0 +1,60 @@
+import 'dart:math';
+
+class SocketJsUtils
+{
+  static final SocketJsUtils _instance = SocketJsUtils._internal();
+
+  factory SocketJsUtils() => _instance;
+
+  SocketJsUtils._internal(); // private constructor
+  
+  final Random _random = Random();
+
+  String generateTransportUrl(String url){
+    var uri = Uri.parse(url);
+
+    var pathSegments = <String>[];
+    if(uri.pathSegments != null){
+      pathSegments.addAll(uri.pathSegments);
+    }
+    pathSegments.add(_generateServerId());
+    pathSegments.add(_generateSessionId());
+    pathSegments.add('websocket');
+
+    uri = Uri(
+      scheme: uri.scheme,
+      host: uri.host,
+      port: uri.port,
+      fragment: uri.fragment,
+      pathSegments: pathSegments
+    );
+
+    var transportUrl = uri.toString();
+    if (transportUrl.startsWith('https'))
+    {
+        transportUrl = 'wss' + transportUrl.substring(5);
+    }
+    else if (transportUrl.startsWith('http'))
+    {
+        transportUrl = 'ws' + transportUrl.substring(4);
+    }
```

Comment:
We probably should throw an exception here if the url did not start with http or https, right?

Reply:
yes, now i throw an exception